### PR TITLE
replaces chaos holoparasite with gaseous holoparasite

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request
replaces chaos holopara with the gaseous holopara (if someone has a better name suggest it)
instead of the fire and hallucinations, it can now expel gas from itself, flooding an area with one of the following:
o2, n2, n2o, co2, h2o, bz or plasma
its punches cause sparks and it regulates the user's temperature and extinguishes them
https://streamable.com/leukv5

## Why It's Good For The Game
chaos holopara is very underwhelming, the hallucinations arent really that good and ss13 fires arent either, also there was no real theme tied around it, what do hallucinations and fire have in common?
this may be more fun and it plays more with the sandbox, lets see!

## Changelog
:cl: fikou
del: chaos holoparasite
add: gaseous holoparasite, it can expel various gases from its body and stabilizes the users temperature
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
fart holoparasite

**Comment by Teleported-Bread** [comment]
Maybe it could release a small amount of BZ along with the other gases so that the hallucination effects remain in some form? If so, the name chaos holoparasite could still apply. Also, maybe let the holoparasite pry open firedoors (if not already a thing) so that the gases they make can be forcefully spread?

**Comment by Jackal-boop** [comment]
seems cool. let us emag it to make fart noises

**Comment by blackdav123** [comment]
could changing gas use a radial menu bound to right click? might be easier than what I assume is a tgui menu from the video

File: code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	miner_fluff_string = span_holoparasite("You encounter... Plasma, the bringer of fire.")
+	toggle_button_type = /atom/movable/screen/guardian/toggle_mode/atmospheric
+	/// Gas being expelled.
+	var/expelled_gas = null
+	/// How much gas we expel per second.
+	var/gas_rate = 10
+	/// Possible gases to expel, with their expelling multiplier.
+	var/static/list/possible_gases = list(
+		/datum/gas/oxygen = 5,
+		/datum/gas/nitrogen = 75, //yes 75 times. overpressurizing is hard!.
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm:20]: 750 moles of nitro a second? what the hell?

File: code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	carp_fluff_string = span_holoparasite("CARP CARP CARP! You caught one! OH GOD, EVERYTHING'S ON FIRE. Except you and the fish.")
+	miner_fluff_string = span_holoparasite("You encounter... Plasma, the bringer of fire.")
+	toggle_button_type = /atom/movable/screen/guardian/toggle_mode/atmospheric
+	/// Gas being expelled.
+	var/expelled_gas = null
+	/// How much gas we expel per second.
+	var/gas_rate = 10
+	/// Possible gases to expel, with their expelling multiplier.
+	var/static/list/possible_gases = list(
+		/datum/gas/oxygen = 5,
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm:19]: 50 moles of oxy is pretty high too but wow

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm:20]: nvm I think I get it now

**Comment by AdipemDragon** [comment]
Maybe a Volatile Holoparasite?

**Comment by moocowswag** [comment]
Finally a holoparasite that synergizes with a job (atmos tators will be able to use atmos holoparasite to make hydrogen)

**Comment by carshalash** [comment]
Where are the snails?

**Comment by Colovorat** [comment]
>what do hallucinations and fire have in common?
that's the thing, nothing. pure chaos

**Comment by BlueMemesauce** [comment]
Really interesting that if a non-antag miner gets a holopara he could give it to an atmos tech and it could have an actually useful job instead of just being in lavaland the whole round

**Comment by Pinkufeck** [comment]
add hydrogen in there :)

**Comment by Pinkufeck** [comment]
Also talk about underwhelming fires, but your version of the holopara only has that and maybe slipping as something special, and this removes the hallucinations, overall nerf.

**Comment by Fikou** [comment]
> Also talk about underwhelming fires, but your version of the holopara only has that and maybe slipping as something special, and this removes the hallucinations, overall nerf.

ur bad

**Comment by ATH1909** [comment]
maybe the host should receive pressure immunity in addition to the fire dousing/temperature regulation?

would it be possible to add a vacuum mode that deletes air instead of generating it, or would that not do much due to the station's air refill systems?

**Comment by Serxule** [comment]
yeah this is gonna remain pretty much unused since almost everything about it is useless other than water vapor for slipping (but thats more of a gimmick than anything)

**Comment by Serxule** [comment]
o2, n2, n2o, co2, are all pointless as it releases so little gas
h2o is just a 16tc slipping gimmick
same as the first reason, plasma is worse than a basic flamethrower and plasma tank, not to add it costs 16tc
temp regulation and extingushing is useless since you're probably gonna be using a modsuit if you're gonna use its one "good" features (plasma)

**Comment by realthomas69** [comment]
Good pr!

**Comment by realthomas69** [comment]
If it had some sort of pressure resistance then that would be swell

**Comment by Shadowflame909** [comment]
Magicians Red became weather report

**Comment by SELFHELL** [comment]
Seems pretty cool... just needs more types of gas

**Comment by YakumoChen** [comment]
Add bz option

File: code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	if(!expelled_gas)
+		return
+	var/datum/gas_mixture/mix_to_spawn = new()
+	mix_to_spawn.add_gas(expelled_gas)
+	mix_to_spawn.gases[expelled_gas][MOLES] = gas_rate * possible_gases[expelled_gas] * delta_time
+	mix_to_spawn.temperature = T20C
+	var/turf/open/our_turf = get_turf(src)
+	our_turf.assume_air(mix_to_spawn)
+
+/mob/living/simple_animal/hostile/guardian/atmospheric/ToggleMode()
```
Comment (dragomagol) [inline code/modules/mob/living/simple_animal/guardian/types/atmospheric.dm:74]: This should have an INVESTIGATE_ATMOS log in it somewhere

File: code/modules/mob/living/simple_animal/guardian/types/gaseous.dm [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	if(picked_gas == "None")
+		expelled_gas = null
+		QDEL_NULL(particles)
+		to_chat(src, span_notice("You stopped expelling gas."))
+		return
+	var/gas_type = gases[picked_gas]
+	if(!picked_gas || !gas_type)
+		return
+	to_chat(src, span_bolddanger("You are now expelling [picked_gas]."))
+	investigate_log("controlled by [key_name(src)] set their gas type to [picked_gas].", INVESTIGATE_ATMOS)
```
Comment (dragomagol) [inline code/modules/mob/living/simple_animal/guardian/types/gaseous.dm:88]:
```suggestion
	investigate_log("set their gas type to [picked_gas].", INVESTIGATE_ATMOS)
```

Comment (dragomagol) [reply code/modules/mob/living/simple_animal/guardian/types/gaseous.dm:88]: Investigate will add key_name(src)

**Pull request merged**